File: Animation/app/src/main/java/com/example/animation/MainActivity.java
package com.example.animation;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button twin, frame;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        frame = findViewById(R.id.frame);
        twin = findViewById(R.id.twin);


        frame.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent Intent = new Intent(MainActivity.this, Frame.class);
                startActivity(Intent);
                overridePendingTransition(R.anim.perehod_frame, R.anim.back_frame);

            }
        });

        twin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, twin.class);
                startActivity(intent);
                overridePendingTransition(R.anim.perehod_twin, R.anim.back_twin);
            }
        });
    }
}

File: PaperDB/app/src/main/java/com/example/paperdb/MainActivity.kt
package com.example.paperdb

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.core.content.ContentProviderCompat.requireContext
import androidx.core.view.get
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import io.paperdb.Paper

class MainActivity : AppCompatActivity() {
    private lateinit var adapter: RecycleAdapter
    private val carItemsList: MutableList<Car> = mutableListOf()
    lateinit var rv :RecyclerView
    lateinit var  buttonAdd : Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        rv = findViewById(R.id.carRV)
        buttonAdd = findViewById(R.id.buttonAdd)
        Paper.init(this)


        buttonAdd.setOnClickListener{
            intent = Intent(this, AddActivity::class.java);
            startActivity(intent)
        }

        setUpAdapter()
    }

    private fun setUpAdapter() {

        carItemsList.addAll(PaperDb.getCar())
        adapter = RecycleAdapter(this, carItemsList,
            {startActivity(Intent(this, UpdateAndDelete :: class.java).apply {
                putExtra("name", it?.name)
                putExtra("model", it?.model)
            })})

        rv.layoutManager = LinearLayoutManager(this)
        rv.setHasFixedSize(true)
        rv.adapter = adapter
    }
}
